sherpa/astro/data.py

```diff
@@ -3468,6 +3576,9 @@ def _dynamic_group(self, group_func, *args, **kwargs):
         If group_func is a callable then it must return the grouping
         and quality arrays for the new scheme.
 
+        Support for group_func being a callable has been marked as
+        deprecated in 4.17.0 and will be removed in the next release.
```

Comment:
Can we say "may be removed" since I'm not convinced yet?

Reply:
I can just remove this comment since this is really an internal routine so anyone calling it has to be aware of it being changed randomly!

Reply:
I've removed all the comments about removing the callable option.